### ISSUE TITLE
[next]magit-ido-completing-read returns empty string when typing only RET

### ISSUE THREAD

**Issue by npostavs** [magit/magit#1781]
Followup on [discussion in #1750][1750].

> But regardless since completing-read returns the empty string on empty input, magit-completing-read should too, and so all possible magit-completing-read-functions have to so too (or be patched to do).

I would say that even with the "new" style, `magit-ido-completing-read` does return empty string on empty input, but "empty input" for ido means `C-j`. Therefore, in magit's case, `enable-old` is **not** fixing a broken completion function, but it's rather a kludge for broken users who don't know about `C-j`. ;-)

So IMO it suffices to just do:

```diff
diff --git a/magit-utils.el b/magit-utils.el
index a66d204..b1cb542 100644
--- a/magit-utils.el
+++ b/magit-utils.el
@@ -288,8 +288,8 @@ (defun magit-ido-completing-read
          (ido-mode t)
          (ido-ubiquitous-mode t)
          ;; Keep `completing-read-ido' from falling back to regular
-         ;; `completing-read' and force use of `enable-old' style.
-         (ido-ubiquitous-next-override 'enable-old)
+         ;; `completing-read'.
+         (ido-ubiquitous-next-override 'enable)
          (reply (completing-read-ido
                  prompt
                  ;; Unlike `completing-read', `ido-completing-read'
```

[1750]: https://github.com/magit/magit/pull/1750#issuecomment-87316303

**Comment by DarwinAwardWinner** [comment]
> but it's rather a kludge for broken users who don't know about C-j. ;-)

Yep, pretty much. :)

**Comment by tarsius** [comment]
> it's rather a kludge for broken users who don't know about `C-j`. ;-)

I guess I deserved that :-) I'll look into this some more and also comment on the things @DarwinAwardWinner said elsewhere. Might take me a while until I get to it though.

**Comment by npostavs** [comment]
> some old packages present a prompt of the form:

Yes, sure, but magit isn't some old package, so we don't have to worry about it here.

> Might take me a while until I get to it though.

No worries, so far the above patch is working well for me.

**Comment by tarsius** [comment]
A fix is cooking, see https://gitter.im/DarwinAwardWinner/ido-ubiquitous and https://github.com/DarwinAwardWinner/ido-ubiquitous/issues/64.

**Comment by tarsius** [comment]
@npostavs I have added a separate `magit-branch-unset-upstream`,
@DarwinAwardWinner which also means that magit now only requires `ido-completing-read+`.

**Comment by DarwinAwardWinner** [comment]
Ok, the ido-ubiquitous release has happened. Feel free to merge.

**Comment by tarsius** [comment]
Thanks @DarwinAwardWinner.
I have merged #1812 just now.
